### PR TITLE
Fix DDP training hangs when num_denoising > 0 and batch contains zero ground-truth targets 

### LINKED ISSUE #672: DDP training hangs when `num_denoising > 0` and batch contains zero ground-truth targets
### Describe the bug

Multi-GPU DDP training hangs (GPU 100% util, no training progress) when `num_denoising > 0` and some GPUs receive a batch where all images have zero annotations. The hang occurs at the first `backward()` — DDP waits indefinitely for gradient synchronization.

### To Reproduce

1. Use a dataset with a high proportion of empty images (e.g., LEVIR-Ship, ~49% images have no objects)
2. Train with DDP (2+ GPUs)
3. Training hangs immediately, GPU util stays at 100%, no loss is printed

### Expected behavior

Training should proceed normally regardless of empty-image batches.

### Root Cause

In `get_contrastive_denoising_training_group()` (`src/zoo/rtdetr/denoising.py`), when all targets in the local batch are empty, the function returns early:

```python
max_gt_num = max(num_gts)
if max_gt_num == 0:
    return None, None, None, None
```

This causes different computation graphs across DDP processes:

| | GPU 0 (empty batch) | GPU 1 (has targets) |
|---|---|---|
| `dn_meta` | `None` | `{dn_positive_idx, ...}` |
| `dn_aux_outputs` in model output | **absent** | **present** |
| `denoising_class_embed` in graph | **not in graph** | **in graph** |
| gradient of `denoising_class_embed` | `None` (never computed) | computed |

DDP's gradient `all_reduce` waits for `denoising_class_embed`'s gradient from GPU 0, which never arrives → **deadlock**.

Setting `find_unused_parameters=True` does NOT resolve this in certain PyTorch versions due to the conditional branching in the forward pass.

### Confirmed by

- Single-GPU training: ✅ works
- Multi-GPU + `num_denoising: 0`: ✅ works
- Multi-GPU + AI-TOD dataset (few empty images): ✅ works
- Multi-GPU + LEVIR-Ship + `num_denoising: 0`: ✅ works
- Multi-GPU + LEVIR-Ship + default `num_denoising`: ❌ hangs

### Fix

**File**: `src/zoo/rtdetr/denoising.py`

**Change**: Instead of returning `None` when `max_gt_num == 0`, set `max_gt_num = 1` so the function continues to produce valid (but zero-contribution) denoising queries, keeping the computation graph identical across all GPUs.

```diff
     max_gt_num = max(num_gts)
+    is_empty_batch = False
     if max_gt_num == 0:
-        return None, None, None, None
+        max_gt_num = 1
+        is_empty_batch = True

     ...
     input_query_bbox = torch.zeros([bs, max_gt_num, 4], device=device)
+    if is_empty_batch:
+        input_query_bbox[..., :2] = 0.5   # cx, cy = image center
+        input_query_bbox[..., 2:] = 0.1   # w, h = small valid size
```

The dummy queries are masked by `pad_gt_mask=0`, contributing **zero loss** and thus **no impact on convergence**.

### Additional notes

The problem is most easily triggered with datasets having a high proportion of empty images (e.g., LEVIR-Ship: 1135/2320 = 48.9% empty). With batch_size=2 per GPU, ~24% of steps will have all-empty batches on one GPU, triggering the hang.

### PR DESCRIPTION
**Pull request by jeiva2000**
Fix DDP deadlock/mismatch when batch has zero GT targets and num_denoising>0

When all images in a rank's local batch have zero annotations,
get_contrastive_denoising_training_group() returned None early,
excluding the denoising module from that rank's computation graph.
Other ranks with annotated images still ran the module, causing
a DDP gradient all_reduce mismatch (and in some cases a hang).

Fix: instead of returning None, use a single masked dummy query
(pad_gt_mask=0) so the graph stays symmetric across ranks while
contributing zero loss.

Fixes #672

### CODE REVIEW

File: rtdetrv2_pytorch/src/zoo/rtdetr/denoising.py [hunk 1]
```diff
@@ -21,20 +21,47 @@ def get_contrastive_denoising_training_group(targets,
...
+        # symmetric across all ranks.
+        # See: https://github.com/lyuwenyu/RT-DETR/issues/672
+        max_gt_num = 1
+        is_empty_batch = True
+
+    # For the dummy case, force a single group: there is no benefit in
+    # generating num_denoising groups out of one masked-out query, and it
+    # would needlessly inflate tgt_size / attn_mask.
+    if is_empty_batch:
+        num_group = 1
```
Comment (lyuwenyu) [inline rtdetrv2_pytorch/src/zoo/rtdetr/denoising.py]: merge this if/else block into above `if max_gt_num == 0:` block, keep one if/else?

Comment (jeiva2000) [reply rtdetrv2_pytorch/src/zoo/rtdetr/denoising.py]: Good call, done — merged into a single if/else block.

**Review by lyuwenyu — APPROVED** [review]
